### PR TITLE
Implemented the list class in C#

### PR DESCRIPTION
**Pull request by OjewaleKehinde**
Added extra methods(InsertRange, Count, AddRange)

### CODE REVIEW

File: list/C#/List.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            endIndex--;
+            }
+        }   
+
+    //returns the number of items in the list
+    public int Count(){
+        return currentIndex;
+        }
+
+    //Makes the list object iterable
```
Comment (oyekanmiayo) [inline list/C#/List.cs:180]: Hey, please remove these comments.

File: list/C#/List.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
+/*
```
Comment (oyekanmiayo) [inline list/C#/List.cs:1]: I don't see a way to test this. I missed that out with some previous implementations, but we should fix that going forward. 

Implement a main method in this file that calls your implementation to ensure it is what we expect. Please consider different edge cases.

File: list/java/List.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /**
  * @author: Ayomide Oyekanmi
  */
+
```
Comment (oyekanmiayo) [inline list/java/List.java:4]: Please remove this new line.

**Comment by OjewaleKehinde** [comment]
Alright. Noted. I'll add a main method to test it. Should I add print
statements to print the results of each tests case, or I should only
include the code for the tests?

**Comment by oyekanmiayo** [comment]
I'm not sure what you mean by "I should only include the code for the tests". Ideally, you'll do something like


```java
List<Integer> list = new List();
list.add(2) // {2}
list.add(0, 50) // {50, 2}

... and so on
```

I should also be able to copy out your code, paste it into Replit and run it immediately (please ensure that is so :)).

Nit: Please use a linter to format this file.
NB: I haven't looked at your actual implementation yet; I'll do that later :). So, expects comments around that when I do.

**Comment by OjewaleKehinde** [comment]
Alright. I'll do that. What I was trying to ask was whether I should add
print statements that display the items in the list to show that the
methods perform as expected. For example, after List.Add(2) and
List.Add(5), should I print the items in the resulting list to indeeed show
that the items were added at the right indexes? Or should I just stop at
calling the implementation of the methods, without printing to see what the
items in the list are. Please let me know if my question is still unclear.

**Comment by oyekanmiayo** [comment]
As I showed in my previous comment, you only need to show in the comments beside the call what the state of the list is after the call.

**Comment by OjewaleKehinde** [comment]
I have added the test cases. Kindly review. I have also removed the new line from the java file.

**Comment by oyekanmiayo** [comment]
Pls remove the LinkedList implementations from this PR. 

It'll make the PR too big and a PR should focus on just one thing. Open another PR for them

**Comment by oyekanmiayo** [comment]
Oh and it looks like there's another PR for C#, so that'll take precedence.

**Comment by oyekanmiayo** [comment]
I (or Abdulazeez) will review the other PR and you can add to that if you want (for the LLs). 

Although, on slack what I meant was that you should create a PR to update the README for LL with the possible optimizations you proposed. We can talk about it on slack before you do that, if you're interested.

**Comment by OjewaleKehinde** [comment]
Alright. Should I send you a message on Slack  then?

**Comment by OjewaleKehinde** [comment]
> I (or Abdulazeez) will review the other PR and you can add to that if you want (for the LLs).
> 
> Although, on slack what I meant was that you should create a PR to update the README for LL with the possible optimizations you proposed. We can talk about it on slack before you do that, if you're interested.

Alright. Should I send you a message on slack then?

**Comment by OjewaleKehinde** [comment]
Okay

**Pull request closed without merging**